### PR TITLE
#5702 Ability to disable submissions in OJS3

### PR DESCRIPTION
**Pull request by salmanm2003**
'**Enabling/Disabling Submissions**' feature has been added as been discussed here:
https://github.com/pkp/pkp-lib/issues/5702

### CODE REVIEW

File: controllers/grid/settings/sections/form/SectionForm.inc.php [hunk 1]
```diff
@@ -55,6 +55,7 @@ function initData() {
 				'title' => $section->getTitle(null), // Localized
 				'abbrev' => $section->getAbbrev(null), // Localized
 				'reviewFormId' => $section->getReviewFormId(),
+				'isInactive' => $section->getIsInactive(), // #2066: Inverted
```
Comment (NateWr) [inline controllers/grid/settings/sections/form/SectionForm.inc.php:58]: The comment can be removed from this line.

File: controllers/grid/settings/sections/form/SectionForm.inc.php [hunk 2]
```diff
@@ -164,6 +178,6 @@ function execute(...$functionArgs) {
 		// Update section editors
 		$this->_saveSubEditors($journal->getId());
 
-		return parent::execute(...$functionArgs);
+		return parent::execute(...$functionArgs);
```
Comment (NateWr) [inline controllers/grid/settings/sections/form/SectionForm.inc.php:181]: Looks like some whitespace was added here.

File: locale/en_US/manager.po [hunk 3]
```diff
@@ -266,6 +266,21 @@ msgstr "Items can only be submitted by Editors and Section Editors."
 msgid "manager.sections.editors"
 msgstr "Section editors"
 
+msgid "manager.sections.inactive"
+msgstr "Inactive"
```
Comment (NateWr) [inline locale/en_US/manager.po:270]: Let's change this to `common.inactive` and move it to pkp-lib's `common.po` file. I suspect we'll get more use out of this in the future.

File: locale/en_US/manager.po [hunk 4]
```diff
@@ -266,6 +266,21 @@ msgstr "Items can only be submitted by Editors and Section Editors."
...
+
+msgid "manager.sections.form.deactivateSection"
+msgstr "Deactivate section (disables new submissions)"
+
+msgid "manager.sections.confirmDeactivateSection"
+msgstr "Are you sure you wish to deactivate this section?"
+
+msgid "manager.sections.confirmActivateSection"
+msgstr "Are you sure you wish to activate this section?"
```
Comment (NateWr) [inline locale/en_US/manager.po:279]: Any locale keys that don't use OJS-specific terminology, like "journal" or "article", can be moved to pkp-lib. That way they can be re-used in OMP/OPS.

File: pages/management/SettingsHandler.inc.php [hunk 5]
```diff
@@ -44,7 +44,11 @@ function __construct() {
 	 */
 	function workflow($args, $request) {
 		parent::workflow($args, $request);
-		TemplateManager::getManager($request)->display('management/workflow.tpl');
+
+		$templateMgr = TemplateManager::getManager($request);
+		$context = $request->getContext();
+		
+		$templateMgr->display('management/workflow.tpl');
```
Comment (NateWr) [inline pages/management/SettingsHandler.inc.php:51]: I _think_ this code is unnecessary...

File: controllers/grid/settings/sections/SectionGridCellProvider.inc.php [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+				$router = $request->getRouter();
+				import('lib.pkp.classes.linkAction.LinkAction');
+
+				if ($element['inactive']) {
+					return array(new LinkAction(
+						'activateSection',
+						new RemoteActionConfirmationModal(
+							$request->getSession(),
+							__('manager.sections.confirmActivateSection'),
```
[the remark this answers is not in the document]
Comment (salmanm2003) [reply controllers/grid/settings/sections/SectionGridCellProvider.inc.php:53]: I can see both loaded when testing, not sure if it is the cache?

> This locale key and the one below (`manager.sections.confirmDeactivateSection`) are not loaded so you'll need to use `AppLocale::requireComponents()`.

File: templates/management/context.tpl [hunk 7]
```diff
@@ -20,6 +20,13 @@
 		</notification>
 	{/if}
 
+	{if !$currentContext->getData('enableSubmissions')}
+		<notification>
+			{capture assign=url}{url page="management" op="settings" path="workflow" anchor="submission/allowSubmissions"}{/capture}
+			{translate key="manager.setup.allowSubmissions.notAccepting"}
```
[the remark this answers is not in the document]
Comment (salmanm2003) [reply templates/management/context.tpl:26]: author.submit.notAccepting is missing "Visit the workflow settings to allow submissions.".
If I have to use it, then I need to add the above part.

File: controllers/grid/settings/sections/SectionGridCellProvider.inc.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (NateWr) [reply controllers/grid/settings/sections/SectionGridCellProvider.inc.php:53]: Hmm, I think there were a couple commits after I started the last code review, so I'll check this again on my next review.

File: templates/management/context.tpl (hunk 7 above)
[the remark this answers is not in the document]
Comment (NateWr) [reply templates/management/context.tpl]: :+1:

File: classes/services/SectionService.inc.php [hunk 8]
```diff
@@ -24,19 +24,22 @@ class SectionService implements EntityPropertyInterface {
 	 * Get array of sections
 	 *
 	 * @param int $contextId
+	 * @param boolean $activeOnly
```
[the remark this answers is not in the document]
Comment (salmanm2003) [reply classes/services/SectionService.inc.php]: > Minor thing, but do you mind providing a description? Something like "Exclude inactive sections from the section list that is returned".

Is this been resolved, as I can see the description in the new release?

**Pull request merged**